lint-staged.config.js

```diff
@@ -7,7 +7,20 @@ module.exports = {
   '*.ctp.graphql': () => 'yarn generate-types:ctp',
   '*.settings.graphql': () => 'yarn generate-types:settings',
   '*.proxy.graphql': () => 'yarn generate-types:proxy',
-  '*.{js,ts,tsx}': [
+  '*.{js}': [
+    'yarn prettier --write',
+    // NOTE: apparently if you pass some argument that is not a flag AFTER the `reporters`
+    // flag, jest does not seem correctly parse the arguments.
+    //
+    //   No tests found related to files changed since last commit.
+    //   Run Jest without `-o` or with `--all` to run all tests.
+    //   Error: An error occurred while adding the reporter at path "/path/to/file".Reporter is not a constructor
+    //
+    // For that reason, we move the `--onlyChanged` flag next to it.
+    'yarn lint:js --reporters=jest-silent-reporter --onlyChanged',
+    'tsc-files --noEmit',
+  ],
+  '*.{ts,tsx}': [
```

Comment:
Isn't `'*.{js}'` now exactly the same as the ` '*.{ts,tsx}'` config? Why splitting?

Reply:
Using `tsc-files` on JS files does not really work. So by splitting it we can run `tsc-files` only on actual TS files.

But I actually fixed that in another PR: https://github.com/commercetools/merchant-center-application-kit/pull/2313/files#diff-29c15f0b5d8d1144bd22153c302cde16c2de092d65b122a618cbdb3023336346

Let me revert the change here to not mix things up. Thanks for bringing this up.